### ISSUE TITLE
fs.watch handlers not cleared after page reload

### ISSUE THREAD

**Issue by fpoli** [nwjs/nw.js#3895]
I'm using [fs.watch](https://nodejs.org/docs/latest/api/fs.html#fs_fs_watch_filename_options_listener) in my application, and I found that after a page reload (triggered using the left button in the toolbar or via `window.location.reload()`) the old event handlers continue to be called.

I expect that after a page reload the app starts from scratch, but apparently it doesn't. Is there a way to perform a "hard reload" without closing the app?

I'm using `Ubuntu 14.04.3 LTS`. nw.js version is:

```
nw.js v0.12.3
io.js v1.2.0
Chromium 41.0.2272.76
commit hash: 591068b-b48a69e-27b6800-459755a-2bdc251-1764a45
```
### Example:

``` html
<!DOCTYPE html>
<html>
  <head>
<script>

var creation = Date.now();

function fileEventHandler(event, name) {
  console.log("fileEventHandler created at "+ creation + ":", event, name);
}

var fs = require("fs");

fs.watch("./", fileEventHandler);

</script>
  </head>
  <body>
    <h1>Hello World</h1>
    <button onclick="window.location.reload()">Reload</button>
  </body>
</html>
```

After 5 page reloads, 6 handlers are called for the same file event instead of one.

```
fileEventHandler created at 1444841950336: change main.html
fileEventHandler created at 1444841961575: change main.html
fileEventHandler created at 1444841961839: change main.html
fileEventHandler created at 1444841962074: change main.html
fileEventHandler created at 1444841962349: change main.html
fileEventHandler created at 1444841962596: change main.html
```
### clearCache

After reading [Livereload nw.js on changes](https://github.com/nwjs/nw.js/wiki/Livereload-nw.js-on-changes) I also tried adding this at the beginning of the script, but nothing changes.

``` javascript
// Clear stuff
for(var module in require.cache){
    if(require.cache.hasOwnProperty(module)){
        delete require.cache[module];
    }
}
var gui = require("nw.gui");
gui.App.clearCache();
```
##

**Comment by fpoli** [comment]
`Window.reloadDev()` seems to work.

**Comment by b1rdex** [comment]
Right refresh button on toolbar supposed to do `Window.reloadDev()`.

**Comment by fpoli** [comment]
Yes, the right refresh button works. Thank you!
I leave the issue open because I think it's a bug that those event handlers are not cleared
- when changing page by clicking on a link
- when refreshing the page with the left refresh button or with `window.location.reload()`

**Comment by Christywl** [comment]
@fpoli , I try to reproduce this issue with your code on the latest nwjs build v0.22.3, but the event handler is not triggered when I click 'Reload' button. There is no log in the devtools console.

**Comment by fpoli** [comment]
@Christywl the event handler is triggered by `fs.watch("./", ...)` when a file changes in the "./" folder. Have you tried adding/deleting/changing a file in "./" while the app is open? For example, with the command `touch main.html`.

To sum up: `fileEventHandler` should be called exactly once per filesystem event, but after a few page reloads (`Window.reload`, not `Window.reloadDev`) `fileEventHandler` seems to be called multiple times per filesystem event. The last time I checked this was in 2015, perhaps it has been fixed in the meantime.
